authentication/serializers.py

```diff
@@ -0,0 +1,38 @@
+from rest_framework import serializers
+from .utils import FirebaseAPI
+from .models import User, UserProfile
+from rest_framework.exceptions import ParseError
+from drf_yasg.utils import swagger_serializer_method
+
+class ResponseSerializer(serializers.Serializer):
+    token = serializers.CharField(max_length=500)
+
+class LoginSerializer(serializers.Serializer):
+    id_token = serializers.CharField(max_length=2400)
+
+    def validate_access_token(self, access_token):
+        return FirebaseAPI.verify_id_token(access_token)
+
+    def validate(self, data):
+        id_token = data.get('id_token', None)
+        user = None
+        jwt = self.validate_access_token(id_token)
```

Comment:
@nishantwrp I don't think it is necessary to add a try-except block here. This line will call the function `validate_access_token()` defined above, which has a try-except block, so that function will raise `ValidationError` if there is any `500` error from Firebase.